app/scripts/lib/local-store.js

```diff
@@ -88,3 +88,14 @@ module.exports = class ExtensionStore {
 function isEmpty (obj) {
   return Object.keys(obj).length === 0
 }
+
+/**
+ * Returns an Error if extension.runtime.lastError is present
+ * this is a workaround for the non-standard error object thats used
+ * @returns {Error}
+ */
+function checkForError () {
+  const lastError = extension.runtime.lastError
+  if (!lastError) return
+  return new Error(lastError.message)
```

Comment:
Perhaps we should check whether it's already an error first, before extracting the message to construct a new error?

I haven't tested this on Firefox, but MDN does claim that `runtime.lastError` is an Error object. So maybe an `instanceof Error` check would catch that case, preserving any other metadata that error might have (like a stack trace).